component/distributed_cache/distributed_cache.go

```diff
@@ -400,6 +401,12 @@ func (dc *DistributedCache) StreamDir(options internal.StreamDirOptions) ([]*int
 		if isMountPointRoot(rawPath) {
 			dirList = hideCacheMetadata(dirList)
 		}
+
+		// Retry listing when the list has no dirents after hiding the root cache dir.
+		if (len(dirList) == 0) && token != "" {
+			options.Token = token
+			goto listAzurePath
+		}
```

Comment:
this is a common condition we must ensure before we exit the function, i.e., we should never exit the function with an empty dirList and not-empty token.
So, move L477-486 to L488 and then simply goto L383.

This way we will never exit with that buggy condition.